src/invidious/channels.cr

```diff
@@ -41,13 +41,13 @@ struct ChannelVideo
     end
   end
 
-  def to_xml(locale, host_url, xml : XML::Builder)
+  def to_xml(locale, host_url, query_params, xml : XML::Builder)
     xml.element("entry") do
       xml.element("id") { xml.text "yt:video:#{self.id}" }
       xml.element("yt:videoId") { xml.text self.id }
       xml.element("yt:channelId") { xml.text self.ucid }
       xml.element("title") { xml.text self.title }
-      xml.element("link", rel: "alternate", href: "#{host_url}/watch?v=#{self.id}")
+      xml.element("link", rel: "alternate", href: "#{host_url}/watch?v=#{self.id}&#{query_params}")
```

Comment:
This actually can be made a little simpler. Something like:

```diff
diff --git a/src/invidious/channels.cr b/src/invidious/channels.cr
index 6004476..cce9831 100644
--- a/src/invidious/channels.cr
+++ b/src/invidious/channels.cr
@@ -44,0 +45,2 @@ struct ChannelVideo
+    query_params["v"] = self.id
+
@@ -50 +52 @@ struct ChannelVideo
-      xml.element("link", rel: "alternate", href: "#{host_url}/watch?v=#{self.id}&#{query_params}")
+      xml.element("link", rel: "alternate", href: "#{host_url}/watch?#{query_params}")
@@ -59 +61 @@ struct ChannelVideo
-          xml.element("a", href: "#{host_url}/watch?v=#{self.id}") do
+          xml.element("a", href: "#{host_url}/watch?#{query_params}") do
, href: "#{host_url}/watch?#{query_params}") do
```

Which will also fix the dangling `&` when `params` is empty.

Reply:
good idea, implemented!